Mathlib/Topology/ContinuousOn.lean

```diff
@@ -83,6 +83,18 @@ theorem mem_nhdsWithin_iff_exists_mem_nhds_inter {t : Set α} {a : α} {s : Set
     t ∈ 𝓝[s] a ↔ ∃ u ∈ 𝓝 a, u ∩ s ⊆ t :=
   (nhdsWithin_hasBasis (𝓝 a).basis_sets s).mem_iff
 
+/-- If `L` and `R` are neighborhoods of `b` within sets whose union is `Set.univ`, then
+`L ∪ R` is a neighborhood of `b`. -/
+theorem union_mem_nhds_of_mem_nhdsWithin {b : α}
```

Comment:
Can you formulate this as the supremum of two the `nhdsWithin` filters equals the `nhds` filter (c.f. [nhdsWithin_union](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ContinuousOn.html#nhdsWithin_union))

Reply:
Thanks, how about this, which uses `nhdsWithin_union`:
```
theorem union_mem_nhds_of_mem_nhdsWithin {b : α}
    {I₁ I₂ : Set α} (h : Set.univ = I₁ ∪ I₂)
    {L : Set α} (hL : L ∈ nhdsWithin b I₁)
    {R : Set α} (hR : R ∈ nhdsWithin b I₂) : L ∪ R ∈ nhds b := by
  rw [← nhdsWithin_univ b, h, nhdsWithin_union]
  exact ⟨mem_of_superset hL (by aesop), mem_of_superset hR (by aesop)⟩
```
Alternatively, could make the first line of the proof a separate theorem:
```
theorem nhdsWithin_union_univ (b : α)
    {I₁ I₂ : Set α} (hI : Set.univ = I₁ ∪ I₂) :
    nhds b = nhdsWithin b I₁ ⊔ nhdsWithin b I₂ := by
  rw [← nhdsWithin_univ b, hI, nhdsWithin_union]
```